code/game/objects/structures/beds_chairs/bed.dm

```diff
@@ -234,3 +234,17 @@
 	name = "double bed"
 	desc = "A luxurious double bed, for those too important for small dreams."
 	icon_state = "bed_double"
+	buildstackamount = 4
+	max_buckled_mobs = 2
+	///The mob who buckled to this bed second, to avoid other mobs getting pixel-shifted before he unbuckles.
+	var/mob/living/goldilocks
```

Comment:
This looks like it could hard-del. Is `post_unbuckle_mob` called on mob delete, which would clear this?

If not, you should use weakrefs.

Reply:
Both ends make sure there's no hard del with buckling.
https://github.com/tgstation/tgstation/blob/57807dd08c66036f402754b8c536d7312dee2231/code/modules/mob/living/living.dm#L36-L37
https://github.com/tgstation/tgstation/blob/19a89870f113c02c4c80967a4ad4e0d3e0b52970/code/game/atoms_movable.dm#L112